### DISCUSSION TITLE
getting g++ 13.00.00 error and header file errors in cpp code

### DISCUSSION THREAD

**Discussion by Aarnav-JP** [Homebrew/discussions#3839] [Everyday usage]
Hi everyone ,I am getting this error in my code since the recent Mac Ventura update. please help me resolve it - 
`g++: warning: could not understand version '13.00.00'
In file included from /opt/homebrew/Cellar/gcc@11/11.3.0/include/c++/11/aarch64-apple-darwin21/bits/stdc++.h:33,
                 from prac.cpp:11:
/opt/homebrew/Cellar/gcc@11/11.3.0/include/c++/11/cassert:44:10: fatal error: assert.h: No such file or directory
   44 | #include <assert.h>
      |          ^~~~~~~~~~
compilation terminated.`

here is my code for reference - 
` #include <bits/stdc++.h>
   using namespace std;

  #define endl "\n"
  
  typedef long long int ll;
  typedef unsigned long long int ull;
  typedef long double lld;
  void solve()
  {
      cout<<"hello";
  }

  int main()
  {
  #ifndef ONLINE_JUDGE
     // freopen("input.txt", "r", stdin);
      // freopen("output1.txt", "w", stdout);
      freopen("Error.txt", "w", stderr);

#endif
    ll t = 1;

    if (0)
    {
        cin >> t;
    }
    for (; t != 0; --t)
        solve();

    return 0;
 }`

**Comment by p-linnane** [comment]
Please see #3822.

**Reply by Aarnav-JP** [reply]
I installed Xcode 14.1 and other stuff and I am still getting the same warning but with other errors now - 
`g++: warning: could not understand version '13.00.00'
ld: -rpath can only be used when targeting Mac OS X 10.5 or later
collect2: error: ld returned 1 exit status`

**Reply by carlocab** [reply]
Try doing
```
brew update && brew reinstall gcc
```

**Reply by Aarnav-JP** [reply]
Just did that still getting the same error

**Reply by carlocab** [reply]
What is the output of `brew info gcc`, `brew config` and `brew doctor`?

**Reply by Aarnav-JP** [reply]
for `brew info gcc` - 

`==> gcc: stable 12.2.0 (bottled), HEAD
GNU compiler collection
https://gcc.gnu.org/
/opt/homebrew/Cellar/gcc/12.2.0 (1,470 files, 358.9MB) *
  Poured from bottle on 2022-10-27 at 12:04:25
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gcc.rb
License: GPL-3.0-or-later with GCC-exception-3.1
==> Dependencies
Required: gmp ✔, isl ✔, libmpc ✔, mpfr ✔, zstd ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 122,071 (30 days), 445,090 (90 days), 1,571,797 (365 days)
install-on-request: 59,865 (30 days), 217,651 (90 days), 740,856 (365 days)
build-error: 1,040 (30 days)
`

for `brew config` - 

`HOMEBREW_VERSION: 3.6.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6a7eac25e167a1eb2d49e13c8cc530a3188af995
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6babcee437d1aa7495e6a137c7c4970afc8b3d6c
Core tap last commit: 28 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.38.1 => /opt/homebrew/bin/git
Curl: 7.84.0 => /usr/bin/curl
macOS: 13.0-arm64
CLT: 14.1.0.0.1.1666437224
Xcode: 14.1 => /Applications/Xcode_14_1.app/Contents/Developer
Rosetta 2: false
`

for `brew doctor` - 
`Unexpected header files:
.... a lot of header files
`

**Reply by carlocab** [reply]
> for `brew doctor` - 
> 
> `Unexpected header files:
> 
> .... a lot of header files
> 
> `

This seems suspicious.

What's the output of
```
which -a ld
ld -v
xcrun ld -v
```
?

**Reply by Aarnav-JP** [reply]
`/usr/bin/ld
@(#)PROGRAM:ld  PROJECT:ld64-820.1
BUILD 18:42:42 Sep 11 2022
configured to support archs: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h armv6m armv7k armv7m armv7em
LTO support using: LLVM version 14.0.0, (clang-1400.0.29.202) (static support for 29, runtime is 29)
TAPI support using: Apple TAPI version 14.0.0 (tapi-1400.0.11)
@(#)PROGRAM:ld  PROJECT:ld64-820.1
BUILD 18:42:42 Sep 11 2022
configured to support archs: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h armv6m armv7k armv7m armv7em
LTO support using: LLVM version 14.0.0, (clang-1400.0.29.202) (static support for 29, runtime is 29)
TAPI support using: Apple TAPI version 14.0.0 (tapi-1400.0.11)`

**Reply by Aarnav-JP** [reply]
@carlocab any solution please?

**Reply by carlocab** [reply]
Sorry, I just realised you were using `gcc@11`. This does not work on Ventura at the moment. If you need GCC, please use `gcc@12`.

Otherwise, you can try building `gcc@11` from source, but it will require patches to build.

**Reply by Aarnav-JP** [reply]
How do i use gcc@12 please help?

**Reply by carlocab** [reply]
You already have it installed. Use `g++-12` instead of `g++-11`.

**Reply by Aarnav-JP** [reply]
how do I use it as my default coz when I compiled my code using command `g++-12 file.cpp` it compiled fine but when I hit run in my vs code or if run command `g++ -v` in terminal it shows shows it is still on g++11. Can you please tell me how do I change that I guess that will really help.
@carlocab

**Reply by carlocab** [reply]
Sorry, I don't use VS Code so I have no idea. The fact that `g++ -v` calls `g++-11` is a consequence of your local configuration, and I have no idea how to fix that as well.

**Reply by Aarnav-JP** [reply]
okay no problem thank you so much

**Comment by codeakki** [comment]
getting same error

**Reply by Aarnav-JP** [reply]
Did you upgrade your xcode to 14.1?

**Reply by codeakki** [reply]
> Did you upgrade your xcode to 14.1?

Yes😢

**Reply by Aarnav-JP** [reply]
Are you also getting ld -rpath error ?after update?

**Reply by codeakki** [reply]
Exactly same as upper mentioned

**Reply by Aarnav-JP** [reply]
do you have gcc@12 installed ?
I have installed it in my system I just don't how to change change it from 11 to 12. if you installed it and you run g++-12 filename.cpp in your terminal it will work fine it worked for me.

**Reply by codeakki** [reply]
Give me some time i will fixed this up :) will revert you back

**Reply by codeakki** [reply]
even g++-12 giving same error :) for me

**Reply by saranyamaity2000** [reply]
I installed the new CommandLine candidate release still its the same. Just 2-3 weeks back was facing a linking error , and now this. I always regret upgrading to newer version

**Reply by Aarnav-JP** [reply]
Me too man every time any update happens everything goes haywire its really frustrating man.

**Reply by Aarnav-JP** [reply]
> Give me some time i will fixed this up :) will revert you back 
> 
> 
> 
> 

Sure man please do.

**Reply by codeakki** [reply]
Got a temporary solution that works fine but sometimes behaves badly:)i.e setting c++-12 as the default build in vs code

**Comment by saranyamaity2000** [comment]
**Got a parmanent solution.** (worked for me) and **I had the exact same error as the author**.

First please make sure you have the latest command line tool ( Command Line Tools for Xcode 14.1 Release Candidate 2 , download from "https://developer.apple.com/download/all" , after downloading and installing make sure to restart your mac once)

Next, go to terminal and enter the command "**brew install gcc**" 
It will basically make sure you have the latest gcc 12.

then use "**g++ --version**".
Most probably it will show you Apple clang as default.
So then change it to the brew gcc latest version.

To do that, 
use "**ln -s g++-12 g++**"

Now just close the terminal and reopen it. 
Now use the command "**g++ --version**" , you should see the brew version correctly.

Now try to compile c++ code , it will work fine :) I hope it helps

**Reply by ParthJohri** [reply]
Hey, I did the exact steps, but I am still getting `g++: warning: could not understand version '13.00.00'` @saranyamaity2000  @Aarnav-JP
